processing_r/processing/r_templates.py

```diff
@@ -0,0 +1,159 @@
+class RTemplates:
+
+    def __init__(self):
+        self._use_sf = True
+        self._use_raster = True
+
+    @property
+    def use_sf(self):
+        return self._use_sf
+
+    @use_sf.setter
+    def use_sf(self, use: bool):
+        self._use_sf = use
+
+    @property
+    def use_raster(self):
+        return self._use_raster
+
+    @use_raster.setter
+    def use_raster(self, use: bool):
+        self._use_raster = use
+
+    def get_necessary_packages(self):
+        packages = []
+
+        if self.use_sf:
+            packages.append("sf")
+        else:
+            packages.append("rgdal")
+            packages.append("sp")
+
+        if self.use_raster:
+            packages.append("raster")
+        else:
+            packages.append("rgdal")
+
+        return packages
+
+    def __set_variable_vector_sf(self, variable: str, path: str, layer: str = None):
+
+        command = ""
+
+        if layer is not None:
+            command = '{0} <- st_read("{1}", layer = "{2}", quiet = TRUE, stringsAsFactors = FALSE)'.format(variable,
+                                                                                                            path, layer)
+        else:
+            command = '{0} <- st_read("{1}", quiet = TRUE, stringsAsFactors = FALSE)'.format(variable, path)
+
+        return command
+
+    def __set_variable_vector_rgdal(self, variable: str, path: str, layer: str = None):
+
+        command = ""
+
+        if layer is not None:
+            command = '{0} <- readOGR("{1}", layer="{2}")'.format(variable, path, layer)
+        else:
+            command = '{0} <- readOGR("{1}")'.format(variable, path)
+
+        return command
+
+    def set_variable_vector(self, variable: str, path: str, layer: str = None):
+
+        if self.use_sf:
+            return self.__set_variable_vector_sf(variable, path, layer)
+        else:
+            return self. __set_variable_vector_rgdal(variable, path, layer)
+
+    def __set_variable_raster_raster(self, variable: str, path: str):
+        return '{0} <- brick("{1}")'.format(variable, path)
+
+    def __set_variable_raster_gdal(self, variable: str, path: str):
+        return '{0} <- readGDAL("{1}")'.format(variable, path)
+
+    def set_variable_raster(self, variable: str, path: str):
+
+        if self.use_raster:
+            return self.__set_variable_raster_raster(variable, path)
+        else:
+            return self.__set_variable_raster_gdal(variable, path)
+
+    def set_variable_extent(self, variable: str, x_min: float, x_max: float, y_min: float, y_max: float):
+        return '{0} <- extent({1},{2},{3},{4})'.format(variable, x_min, x_max, y_min, y_max)
+
+    def set_variable_string(self, variable: str, value: str):
+        return '{0} <- "{1}"'.format(variable, value)
```

Comment:
Looking at this now - we should probably handle escaping for value too

Reply:
Not sure if it it is really necessary here and maybe more importantly, I dont know how to handle it correctly. Any tips?

Reply:
See https://stat.ethz.ch/R-manual/R-devel/library/base/html/Quotes.html -- We'd need to do something like:

    return '{0} <- "{1}"'.format(variable, value.replace('"', '\\"' )

(i.e. replace `"` characters in value with `\"`)

Reply:
Ok, I see. I did not realized that this could be an issue.  It is now included in the PR.